Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/machine/MachineLocalState.java

```diff
@@ -0,0 +1,23 @@
+package pexplicit.runtime.machine;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Represents the local state of a machine
+ */
+public class MachineLocalState implements Serializable {
+    /**
+     * List of values of all local variables (including internal variables like currentState, FIFO queue, etc.)
```

Comment:
I am a bit worried if this is the best way to track local state.

Reply:
Let's discuss this further offline and revamp this in the next PR.